crates/core/src/model/scalar/mod.rs

```diff
@@ -1036,4 +1044,44 @@ mod test {
         assert_eq!(empty, TRUE);
         assert!(empty.query_state(State::Truthy));
     }
+
+    #[test]
+    fn borrows_from_scalar_cow() {
+        fn is_borrowed(cow: Cow<'_, str>) -> bool {
+            match cow {
+                Cow::Borrowed(_) => true,
+                Cow::Owned(_) => false,
+            }
+        }
+
+        let s: String = "gamma".into();
+        let sc: ScalarCow<'_> = s.into();
+
+        // clones instead of borrowing
+        {
+            fn extract_cow_str<'s>(value: &'s dyn ValueView) -> Cow<'s, str> {
+                value.to_kstr().into_cow_str()
+            }
+            assert_eq!(is_borrowed(extract_cow_str(sc.as_view())), false);
+        }
+
+        // does not compile
+        /* {
+            fn extract_cow_str<'s>(value: &'s dyn ValueView) -> Cow<'s, str> {
+                // error: cannot return value referencing temporary value:
+                // `to_kstr(&'a self)` returns a `KStringCow<'a>`, rather than
+                // a `KStringCow<'s>`
+                value.as_scalar().to_kstr().into_cow_str()
+            }
+            assert_eq!(is_borrowed(extract_cow_str(&sc)), true);
+        } */
```

Comment:
Let's break this down
```rust
            fn extract_cow_str<'s>(value: &'s dyn ValueView) -> Cow<'s, str> {
                // error: cannot return value referencing temporary value:
                // `to_kstr(&'a self)` returns a `KStringCow<'a>`, rather than
                // a `KStringCow<'s>`
                let o = value.as_scalar();
                let k = o.to_kstr();
                k.into_cow_str()
            }
```
`o`s type is
```
error[E0308]: mismatched types
    --> crates/core/src/model/scalar/mod.rs:1075:21
     |
1075 |                 let () = o;
     |                     ^^ expected enum `std::option::Option`, found `()`
     |
     = note:   expected enum `std::option::Option<model::scalar::ScalarCow<'_>>`
             found unit type `()`
```
So if `to_kstr` borrow's `o` then it won't live long enough

Reply:
Whats your preference for how this dead code gets handled before merging?

Reply:
I'm not sure. I wish "compile-fail" tests were a standard Rust feature already, but they're not!

Perhaps it's not useful to have this code at all in there - I wanted you to see it on review though. I can delete it and amend+force-push, or you can drop it yourself, whichever is most convenient for you!

Reply:
If you don't mind deleting and force-pushing, that'd b e great.  Post here when done since github doesn't send notifications for force-pushes.  I'll then merge this and cut a release.